cmd/harmony/main.go

```diff
@@ -701,7 +689,30 @@ func setupConsensusAndNode(hc harmonyconfig.HarmonyConfig, nodeConfig *nodeconfi
 		chainDBFactory = &shardchain.LDBFactory{RootDir: nodeConfig.DBDir}
 	}
 
-	currentNode := node.New(myHost, currentConsensus, chainDBFactory, blacklist, allowedTxs, localAccounts, nodeConfig.ArchiveModes(), &hc)
+	engine := chain.NewEngine()
+
+	chainConfig := nodeConfig.GetNetworkType().ChainConfig()
+	collection := shardchain.NewCollection(
+		&hc, chainDBFactory, &core.GenesisInitializer{NetworkType: nodeConfig.GetNetworkType()}, engine, &chainConfig,
+	)
+
+	blockchain, err := collection.ShardChain(nodeConfig.ShardID)
```

Comment:
Hey, i found the bug. The archival mode is not work after this MR.

Calling `collection.shardChain` before new Node will make `collection.DisableCache(shardID)` (inside node.New) is useless. 

Since `collection.shardChain` need to check to disable cache for archival mode.

```
if sc.disableCache[shardID] {
    cacheConfig = &core.CacheConfig{Disabled: true}
    utils.Logger().Info().
    Uint32("shardID", shardID).
        Msg("disable cache, running in archival mode")
}
```